recipes/reflectcpp/all/conanfile.py

```diff
@@ -0,0 +1,151 @@
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cbor": [True, False],
+        "with_flatbuffers": [True, False],
+        "with_msgpack": [True, False],
+        "with_toml": [True, False],
+        "with_ubjson": [True, False],
+        "with_xml": [True, False],
+        "with_yaml": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cbor": False,
+        "with_flatbuffers": False,
+        "with_msgpack": False,
+        "with_toml": False,
+        "with_ubjson": False,
+        "with_xml": False,
+        "with_yaml": False,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("ctre/3.9.0", transitive_headers=True)
+        # INFO: include/rfl/json/Writer.hpp includes yyjson.h
+        # INFO: Transitive lib needed to avoid undefined reference to symbol 'yyjson_mut_doc_new'
+        self.requires("yyjson/0.10.0", transitive_headers=True, transitive_libs=True)
+        if self.options.with_cbor:
+            self.requires("tinycbor/0.6.0", transitive_headers=True)
+        if self.options.with_flatbuffers:
+            self.requires("flatbuffers/24.3.25", transitive_headers=True)
+        if self.options.with_msgpack:
+            self.requires("msgpack-c/6.0.0", transitive_headers=True)
+        if self.options.with_toml:
+            self.requires("tomlplusplus/3.4.0", transitive_headers=True)
+        if self.options.with_ubjson:
+            self.requires("jsoncons/0.176.0", transitive_headers=True)
+        if self.options.with_xml:
+            self.requires("pugixml/1.14", transitive_headers=True)
+        if self.options.with_yaml:
+            self.requires("yaml-cpp/0.8.0", transitive_headers=True)
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.23 <4]")
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), "set(CMAKE_CXX_STANDARD 20)", "")
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        deps = CMakeDeps(self)
```

Comment:
```suggestion
        deps = CMakeDeps(self)
        if self.options.with_flatbuffers:
            deps.set_property("flatbuffers", "cmake_target_name", "flatbuffers::flatbuffers")
```

Because Conan generates the CMake target `flatbuffers::libflatbuffers`. This change fixes the follow error:


```
-- Configuring done (1.2s)
CMake Error at CMakeLists.txt:131 (target_link_libraries):
  Target "reflectcpp" links to:

    flatbuffers::flatbuffers

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (0.1s)
CMake Generate step failed.  Build files cannot be regenerated correctly.

reflectcpp/0.16.0: ERROR: 
Package 'e71c47e4a8170b93577b9cb78554080767eaafb9' build failed
```